File: tests/agent_workloads/adapters/incident_report.mjs
import { readFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: incident_report INPUT_FILE");
}

const incidents = [];
const rejects = [];
const lines = readFileSync(process.argv[2], "utf8").split(/\r?\n/);
for (let index = 1; index < lines.length; index += 1) {
  const line = lines[index];
  if (!line) continue;
  const fields = line.split("\t");
  const lineNumber = index + 1;
  if (fields.length !== 3) {
    rejects.push(`reject|${lineNumber}|field-count`);
    continue;
  }
  const [service, status] = fields;
  if (status !== "ok" && status !== "error") {
    rejects.push(`reject|${lineNumber}|status`);
  } else if (!service) {
    rejects.push(`reject|${lineNumber}|service`);
  } else {
    incidents.push([service, status]);
  }
}

const rows = [`accepted|${incidents.length}`, `rejected|${rejects.length}`, ...rejects];
const services = [...new Set(incidents.map(([service]) => service))].sort();
for (const service of services) {
  const ok = incidents.filter((item) => item[0] === service && item[1] === "ok").length;
  const errors = incidents.filter(
    (item) => item[0] === service && item[1] === "error",
  ).length;
  rows.push(`${service}|ok=${ok}|error=${errors}`);
}
console.log(rows.join("\n"));

File: tests/agent_workloads/adapters/incident_report.py
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: incident_report INPUT_FILE")

incidents = []
rejects = []
for line_number, line in enumerate(Path(sys.argv[1]).read_text().splitlines()[1:], 2):
    if not line:
        continue
    fields = line.split("\t")
    if len(fields) != 3:
        rejects.append(f"reject|{line_number}|field-count")
        continue
    service, status, _duration = fields
    if status not in {"ok", "error"}:
        rejects.append(f"reject|{line_number}|status")
    elif not service:
        rejects.append(f"reject|{line_number}|service")
    else:
        incidents.append((service, status))

rows = [f"accepted|{len(incidents)}", f"rejected|{len(rejects)}", *rejects]
for service in sorted({service for service, _status in incidents}):
    ok = sum(item == (service, "ok") for item in incidents)
    errors = sum(item == (service, "error") for item in incidents)
    rows.append(f"{service}|ok={ok}|error={errors}")
print(*rows, sep="\n")
